src/main/java/org/mskcc/cmo/metadb/service/impl/ValidRequestCheckerImpl.java

```diff
@@ -0,0 +1,269 @@
+     * If the total number of samples available is equal to valid samples, then the request as a whole is valid
+     * If the number of valid samples is zero, the request is invalid and will be logged
+     * If the number of valid samples is less than the total number of samples, the request is still considered valid but logged to keep note of the invalid samples 
+     * @throws IOException 
+     */
+    @Override
+    public String checkIfValidRequest(String requestJson) throws IOException {
+        
+        if (StringUtils.isAllBlank(requestJson)) {
+            return null;
+        }
+        
+        Map<String, Object> requestJsonMap = mapper.readValue(requestJson, Map.class);
+        isCmoRequest = isCmoRequest(requestJsonMap);
+        hasRequestId = hasRequestId(requestJsonMap);
+        
+        if (igoCmoRequestFilter && !isCmoRequest) {
+            LOG.info("CMO request filter enabled - skipping non-CMO request: "
+                    + requestJsonMap.get("requestId"));
+            return null;
+        }
+        
+        Object[] sampleList = mapper.convertValue(requestJsonMap.get("samples"),
+                Object[].class);
+        List<Object> validSampleList = new ArrayList<Object>();
+        int numOfSamples = sampleList.length;
+        
+        for (Object sample: sampleList) {
+            Map<String, String> sampleMap = mapper.convertValue(sample, Map.class);
+            if(!igoCmoRequestFilter && !isCmoRequest) {
+                if (checkIfValidNonCmoSample(sampleMap)) {
+                    validSampleList.add(sample);
+                }
+            } else {
+                if (checkIfValidCmoSample(sampleMap)) {
+                    validSampleList.add(sample);
+                }
+            }
+            
+        }
+        
+        if (validSampleList.size() > 0) {
+            if(validSampleList.size() < numOfSamples) {
+                String newSamplesString = mapper.writeValueAsString(validSampleList.toArray(new Object[0]));
+                requestJsonMap.replace("samples", newSamplesString);
+                LOG.info("CMO request passed sanity checking - missing samples...");
+                requestStatusLogger.logRequestStatus(requestJson,
+                        RequestStatusLogger.StatusType.REQUEST_WITH_MISSING_SAMPLES);
+            }
+            return mapper.writeValueAsString(requestJsonMap);
+        }
+        LOG.error("Request failed sanity checking - logging request status...");
+        requestStatusLogger.logRequestStatus(requestJson,
+                RequestStatusLogger.StatusType.CMO_REQUEST_FAILED_SANITY_CHECK);
+        return null;
+    }
+
+    /**
+     * Receives a sampleMap
+     * Check if the sample has all the required fields
+     * If RequestId, InvestigatorSampleId, CmoPatientId,
+     * SpecimenType, SampleType or NormalizedPatientId are missing, returns false
+     */
+    boolean checkIfValidCmoSample(Map<String, String> sampleMap) {
```

Comment:
dont forget to add the scope visibility declaration here and in the other private methods added in this class

can we also rename the methods to follow this format? 
`checkIfValidCmoSample` --> `isValidCmoSample`

so similar but a bit different to reflect that these methods will return a boolean